### PR TITLE
refactor: use zod for input validation

### PR DESCRIPTION
**Pull request by asrvd**
used zod to validate/parse input in guestbook

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -49,3 +49,4 @@ This application is licensed under the [Apache-2.0](https://github.com/nexxeln/n
 - [Theo Browne](https://t3.gg/) for gifting the nice [nxl.sh](https://nxl.sh) domain.
 - [Lee Robinson](https://leerob.io/) for inspiration and the README.
 - [Harsh Singh](https://github.com/harshhhdev/) for helping with some of the MDX components.
+- [Ashish](https://github.com/asrvd/) for refactoring guestbook component.
```
Comment (nexxeln) [inline README.md]: for refactoring _the_ guestbook component

**Review by nexxeln — CHANGES_REQUESTED** [review]

**Review by nexxeln — COMMENTED** [review]
Can we also have a custom error message based on what was wrong?

**Comment by asrvd** [comment]
> Can we also have a custom error message based on what was wrong?

whats the current one?

**Comment by asrvd** [comment]
just gave the zod docs a read, its possible, will try it out on my site then add changes here

**Pull request merged**